sycl/doc/extensions/UnnamedKernelLambda/SYCL_INTEL_unnamed_kernel_lambda.asciidoc

```diff
@@ -1,184 +0,0 @@
-
-NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by permission by Khronos.
-
-NOTE: This document is better viewed when rendered as html with asciidoctor.  GitHub does not render image icons.
-
-This document describes an extension that makes it optional, instead of required, to name (through template parameter of invocation function) kernels that are defined as lambdas.
-
-
-== Name Strings
-
-+SYCL_INTEL_unnamed_kernel_lambda+
-
-== Notice
-
-Copyright (c) 2019 Intel Corporation.  All rights reserved.
-
-== Status
-
-Working Draft
-
-This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
-
-Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
-
-== Version
-
-Built On: {docdate} +
-Revision: 1
-
-== Contact
-Michael Kinsner, Intel (michael 'dot' kinsner 'at' intel 'dot' com)
-
-== Dependencies
-
-This extension is written against the SYCL 1.2.1 specification, Revision v1.2.1-5.
-
-== Overview
-
-SYCL allows device kernels to be defined in a variety of ways, one of which is as a lambda.  Separate compilation means that the host-side invocation of a kernel lambda must be matched to the device-compiled variant of the lambda from a potentially different compilation.  Lambda internal compiler naming is an implementation detail and not required to match across different compilers.  Differences in macro expansion and other details between host and device compilations can further lead to divergence of compiled lambda name, for the same lambda in the original source code.
-
-SYCL 1.2.1 requires that a kernel name be specified as a template parameter of the invocation function when a kernel has been defined as a lambda.  This kernel name enables correlation of the lambda name between multiple compilations.
-
-The extension described by this document makes naming of lambdas at invocation time optional in SYCL 1.2.1.  Launch of a kernel is simplified from, for example:
-
-[source,c++,LambdaNamed,linenums]
-----
-...
-cgh.parallel_for<class my_kernel_name>(range<1>{ 1024 }, [=](id<1> idx) {
-  writeResult[idx] = idx[0];
-});
-...
-----
-
-To:
-
-[source,c++,NoName,linenums]
-----
-...
-cgh.parallel_for(range<1>{ 1024 }, [=](id<1> idx) {
-  writeResult[idx] = idx[0];
-});
-...
-----
-
-In this example, `<class my_kernel_name>` is optional and does not need to be included.
-
-== Enabling the extension
-
-In the Intel prototype implementation of this extension at https://github.com/intel/llvm[GitHub link], the extension is enabled through the compiler flag `-fsycl-unnamed-lambda`.  When enabled, the extension defines the macro `$$__SYCL_INTEL_UNNAMED_LAMBDA__$$`.
```

Comment:
Is `-fsycl-unnamed-lambda` documented anywhere else?

Reply:
Yes: https://github.com/intel/llvm/blob/sycl/sycl/doc/UsersManual.md#language-options